modules/ducktests/tests/ignitetest/tests/self_test.py

```diff
@@ -53,3 +57,53 @@ def test_assertion_convertion(self, ignite_version):
         else:
             app.stop()
             assert False
+
+    def test_clock_sync(self):
+        """
+        Tests if clocks are synchronized between nodes.
+        """
+
+        service = GetTimeService(self.test_context, self.test_context.cluster.num_available_nodes())
+        service.start()
+        service.wait(10)
+
+        service.clocks.sort(key=operator.itemgetter(1))
+
+        for _ in service.clocks:
+            self.logger.info("NodeClock[%s] = %d" % (_[0], _[1]))
+
+        max_clock_spread_ms = 100
+        min_res = service.clocks[0]
+        max_res = service.clocks[-1]
+        assert max_res[1] - min_res[1] < max_clock_spread_ms, \
+            "Clock difference between %s (%d) and %s (%d) exceeds %d ms." \
+            % (min_res[0], min_res[1], max_res[0], max_res[1], max_clock_spread_ms)
+
+
+class GetTimeService(BackgroundThreadService):
+    """
+    Collects clock timestamps from a set of nodes.
+    """
+
+    def __init__(self, context, num_nodes):
+        super().__init__(context, num_nodes)
+        self.start_barrier = threading.Barrier(num_nodes)
+        self.clocks = []
+
+    def _worker(self, _, node):
+        node.account.ssh("/bin/true")
+
+        self.start_barrier.wait(1)
+        start = time.time()
+        delay = 5
```

Comment:
What is a purpose of this delay? We can just use `time.time() - start`, as delay is a constant. 

Actually to make a correction we should to run void command, like `ssh ;`.

Reply:
Thanks. It's obsolete now. Removing.